pkg/reconciler/knativeserving/knativeserving.go

```diff
@@ -87,34 +99,76 @@ func (r *Reconciler) FinalizeKind(ctx context.Context, original *servingv1alpha1
 // converge the two.
 func (r *Reconciler) ReconcileKind(ctx context.Context, ks *servingv1alpha1.KnativeServing) pkgreconciler.Event {
 	logger := logging.FromContext(ctx)
+	// Read the old version of the Knative Serving and the version of Knative Serving to be installed
+	oldVerion, newVersion := r.retrieveVersions(ks)
+	if newVersion == "" {
+		logger.Info("No version is specified in Knative Serving CR.")
+		return nil
+	}
+
 	ks.Status.InitializeConditions()
 	ks.Status.ObservedGeneration = ks.Generation
 
 	logger.Infow("Reconciling KnativeServing", "status", ks.Status)
+
+	// Get the Serving Manifest to be installed
+	servingManifest, err := r.retrieveManifest(ctx, newVersion)
+	if err != nil {
+		return err
+	} else if len(servingManifest.Resources()) == 0 {
+		return fmt.Errorf("unable to find the manifest for the Knative Serving version %s", newVersion)
+	}
+
+	// Get the previous Manifest, which has been installed.
+	oldManifest, err := r.retrieveManifest(ctx, oldVerion)
+	if err != nil {
+		return err
+	} else if len(oldManifest.Resources()) == 0 {
+		return fmt.Errorf("unable to find the previous manifest for the Knative Serving version %s", oldVerion)
+	}
+
 	stages := []func(context.Context, *mf.Manifest, *servingv1alpha1.KnativeServing) error{
 		r.ensureFinalizerRemoval,
 		r.install,
 		r.checkDeployments,
+	}
+
+	deleteStages := []func(context.Context, *mf.Manifest, *servingv1alpha1.KnativeServing) error{
 		r.deleteObsoleteResources,
 	}
 
-	manifest, err := r.transform(ctx, ks)
+	oldManifest, err = r.transform(ctx, ks, oldManifest)
+	if err != nil {
+		return err
+	}
+
+	servingManifest, err = r.transform(ctx, ks, servingManifest)
 	if err != nil {
 		ks.Status.MarkInstallFailed(err.Error())
 		return err
 	}
 
+	// Find the common resources between the old and the current serving manifests
+	manifestApply := servingManifest.Filter(mf.In(oldManifest))
```

Comment:
This probably shouldn't filter at all or we'll never create new resources.

Reply:
Done.